### PR TITLE
scroll flaws into view on mount when hash is present

### LINKED ISSUE #605: #show-flaws anchor link broke
### Steps to reproduce:

1. You need to visit at least one page (with flaws) first or run `node content build -l en-us -f web/html`
2. Go to http://localhost:3000/en-US/_flaws
3. Click on any of the listed documents

### Expected result:
That it scrolls you to the bottom where the flaws are listed

### Actual result:
No scrolling happens

I think it broke in
https://github.com/mdn/yari/pull/586/commits/5a2fb413b815d1b4315c4b46a790745daa2f8f7b
which I merged into the big Flaws PR but failed to notice.

### PR DESCRIPTION
**Pull request by Gregoor**
fixes #605

### CODE REVIEW

**Comment by peterbe** [comment]
But why do we need to do that? Aren't we over complicating things? The `id="show-flaws"` is in the DOM so the browser should know "natively" to go there. Why did that stop working?

Where we need code is in that we should sniff, on the initial mount, if the anchor link is present and if so display the flaws.

**Comment by Gregoor** [comment]
The browser only hijacks scrolling on the initial render. Since the element with that hash's ID is rendered later, it is not being scrolled to.
It did sometimes do the right thing before because we forced a browser URL for every toggle, thereby letting the browser hijack scrolling at a time when the referencend element is already rendered.

File: client/src/document/index.tsx [hunk 1]
```diff
@@ -400,13 +401,17 @@ function toggleLocationFlawsHash(on: boolean) {
...
+  const location = useLocation();
+  const [show, toggle] = useReducer((v) => !v, location.hash === FLAWS_HASH);
+  const rootElement = useRef<HTMLDivElement>(null);
+  const isInitialRender = useRef(true);
 
   useEffect(() => {
-    if (window.location.hash && window.location.hash === FLAWS_HASH) {
-      toggle();
+    if (isInitialRender.current && show && rootElement.current) {
+      rootElement.current.scrollIntoView();
```
Comment (peterbe) [inline client/src/document/index.tsx:411]:
```suggestion
      rootElement.current.scrollIntoView({behavior: "smooth", });
```

**Comment by peterbe** [comment]
As discussed, see if you can use `useNavigate()` to update the anchor.

**Pull request merged**